### PR TITLE
feat: Add jaeger support so that syndesis and integrations can be traced.

### PR DESCRIPTION
**Pull request by chirino**
- Add a jaeger based ActivityTrackingService.

### CODE REVIEW

File: app/server/logging/jaeger/pom.xml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      <artifactId>slf4j-api</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>io.fabric8</groupId>
+      <artifactId>openshift-client</artifactId>
+      <scope>runtime</scope>
+    </dependency>
+
+<!--
```
Comment (oscerd) [inline app/server/logging/jaeger/pom.xml]: I think we can remove this if not needed.

File: app/server/logging/jaeger/pom.xml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    </dependency>
+
+    <dependency>
+      <groupId>ch.qos.logback</groupId>
+      <artifactId>logback-classic</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>commons-logging</groupId>
+      <artifactId>commons-logging</artifactId>
+      <scope>test</scope>
```
Comment (zregvart) [inline app/server/logging/jaeger/pom.xml]: Probably should be excluded and replaced with `jcl-over-slf4j`

File: app/server/logging/jaeger/src/main/java/io/syndesis/server/logging/jaeger/JaegerActivityTrackingConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configured in spring.factories so that this configuration is automatically picked
+ * up when included in the classpath.
+ */
+@Configuration
+@ComponentScan
+@ConditionalOnProperty(value = "features.jaeger-activity-tracing.enabled", havingValue = "true")
+public class JaegerActivityTrackingConfiguration {
```
Comment (zregvart) [inline app/server/logging/jaeger/src/main/java/io/syndesis/server/logging/jaeger/JaegerActivityTrackingConfiguration.java]: Needed? I thought we did `@ComponentScan` regardless and it doesn't have any `@Configuration` beans here.

File: app/server/logging/jaeger/src/main/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingService.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            }
+
+        }
+
+        LOG.debug("rc: {}", rc);
+        return rc;
+    }
+
+
+    public static void main(String[] args) throws IOException {
```
Comment (zregvart) [inline app/server/logging/jaeger/src/main/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingService.java]: Remove please

File: app/server/logging/jaeger/src/main/java/io/syndesis/server/logging/jaeger/service/JaegerQueryAPI.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import io.syndesis.common.util.Json;
+import io.syndesis.common.util.SuppressFBWarnings;
+
+/**
+ * Interface to Jaeger's Query API
+ */
+@Component
+public class JaegerQueryAPI {
```
Comment (zregvart) [inline app/server/logging/jaeger/src/main/java/io/syndesis/server/logging/jaeger/service/JaegerQueryAPI.java]: Would be cleaner to use Immutables for those inner DTO classes and place them in a `io.syndesis.server.logging.jaeger.dto` package or similar

File: app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    @Test
+    public void shouldRetainLastRetainActivityLogs() throws IOException {
+
+        // Instead of testing an online service, lets override to use static test data.
+        JaegerActivityTrackingService service = new JaegerActivityTrackingService(new JaegerQueryAPI("http://localhost:16686/api") {
+            @Override
+            public ArrayList<Trace> tracesForService(String service, int lookbackDays, int limit) {
+                try {
+                    String json = resource("example-jaeger-trace-result.json");
+                    ObjectMapper objectMapper = new ObjectMapper();
```
Comment (zregvart) [inline app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java]:
```suggestion
```

File: app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+                    throw new WebApplicationException(e);
+                }
+            }
+        });
+
+        List<Activity> activities = service.getActivities("test", null, null);
+        assertThat(activities).isNotNull();
+
+
+        assertThat(Json.writer().withDefaultPrettyPrinter().writeValueAsString(activities).trim())
```
Comment (zregvart) [inline app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java]: We could use JSONAssert here

File: app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+            .isEqualTo(resource("expected-activities.json").trim());
+
+    }
+
+
+    private static String resource(String file) throws IOException {
```
Comment (zregvart) [inline app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java]: Could use `io.syndesis.common.util.Resources::getResourceAsText` or `io.syndesis.common.util.IOStreams::readText` if using JaegerActivityTrackingServiceTest classloader is required instead

File: app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        ClassLoader loader = JaegerActivityTrackingServiceTest.class.getClassLoader();
+        try (InputStream is = requireNonNull(loader.getResourceAsStream(file))) {
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            copy(is, os);
+            return new String(os.toByteArray(), StandardCharsets.UTF_8);
+        }
+    }
+
+    private static void copy(InputStream is, ByteArrayOutputStream os) throws IOException {
```
Comment (zregvart) [inline app/server/logging/jaeger/src/test/java/io/syndesis/server/logging/jaeger/service/JaegerActivityTrackingServiceTest.java]: Could use `io.syndesis.common.util.IOStreams::copy`

**Pull request merged**